### DISCUSSION TITLE
part-ii/2021-126/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#76] [Question comments]
Part II 2021 Quantum Information and Computation: Paper 2, Section I, 10D 

https://questions.tripos.org/part-ii/2021-126/

**Comment by thenewtonium** [comment]
<details><summary>First part:</summary>

Let $\ket{\eta} := \frac{1}{\sqrt{2^n}} \sum_{x\in B_n} \ket{x}$ and note that
- in case (I), $\ket{\xi} = \ket{\eta}$
- in case (II), $\ket{\xi} \perp \ket{\eta}$

and hence measuring w.r.t. a basis containing $\ket{\eta}$ allows us to determine with certainty if case I or case II holds.
</details>

<details><summary>Second part:</summary>

Alice uses the standard trick (adjoin ancilla qubit in $\ket{-}$ state and then apply oracle $U_{f_A}$ with the ancilla as ‘output’ register) to form $$\ket{\xi _{f_A}} = \frac{1}{\sqrt{2^n}} \sum_{x\in B_n} (-1)^{f_A(x)} \ket{x}$$ from $\ket{\xi}$.

Bob then applies his version of the trick[^BnotA] to the received $\ket{\xi_{f_A}}$ state to yield the state $$\ket{\xi_h} = \frac{1}{\sqrt{2^n}} \sum_{x\in B_n} (-1)^{f_A(x) \oplus f_B(x)} \ket{x}$$.

Case (1) $\iff$ $h$ satisfies (I), and case (2) $\iff$ $h$ satisfies (II). So Bob can now use the first part of the question to determine whether (1) or (2) holds.

[^BnotA]: I.e. with $f_A$ replaced by $f_B$ ofc.

</details>
